pkg/validator/shoot_validator.go

```diff
@@ -126,9 +120,41 @@ func (v *Shoot) validateShootUpdate(ctx context.Context, oldShoot, shoot *core.S
 		}
 	}
 
+	// Only validate against cloud profile when zones are updated.
+	// This ensures that already running shoots won't break after a zone is removed from a cloud profile.
+	if len(infraConfig.Networks.Zones) > len(oldInfraConfig.Networks.Zones) {
+		cloudProfile := &gardencorev1beta1.CloudProfile{}
+		if err := v.client.Get(ctx, kutil.Key(shoot.Spec.CloudProfileName), cloudProfile); err != nil {
+			return err
+		}
+
+		if errList := awsvalidation.ValidateInfrastructureConfigAgainstCloudProfile(infraConfig, shoot, cloudProfile, infraConfigFldPath); len(errList) != 0 {
+			return errList.ToAggregate()
+		}
```

Comment:
Nit: Can this be put into a separate function to prevent replication at lines 150-157?

Otherwise lgtm